### PR TITLE
Add changes to support vck190-pci platform

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
> Added minimal changes to bring up vck190 pci platform
> Added dynamic resources structure for platform in device.h file
> As vsec support is not there in platform hardcoded uuid rom value as discussed with @sonals

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c [hunk 1]
```diff
@@ -1121,8 +1121,12 @@ static void xclmgmt_extended_probe(struct xclmgmt_dev *lro)
 	 * like if versal has vesc, then it is a 2.0 shell. We can add the following
 	 * condition.
 	 */
+
+	/* XOCL_DSAFLAG_CUSTOM_DTB is used for non-VSEC platforms which still wanted to
+	 * use partition metadata to discover resources
+	 */
 	if ((dev_info->flags & (XOCL_DSAFLAG_VERSAL | XOCL_DSAFLAG_MPSOC)) &&
-	    xocl_subdev_is_vsec(lro))
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c:1125]: It seems the logic here is that your device.h flags has to have 2 flags (VERSAL and _DTB)? 
We possible need some code refactoring here, but seems working for me for now.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c [hunk 2]
```diff
@@ -679,6 +679,15 @@ static int get_header_from_vsec(struct feature_rom *rom)
 
 	ret = xocl_subdev_vsec(xdev, XOCL_VSEC_UUID_ROM, &bar, &offset, NULL);
 	if (ret) {
+		/* XOCL_DSAFLAG_CUSTOM_DTB is used for non-VSEC platforms which
+		 * still wanted to use partition metadata to discover resources
+		 */
+		if (XDEV(xdev)->priv.flags & XOCL_DSAFLAG_CUSTOM_DTB) {
+			rom->uuid_len = 32;
+			strcpy(rom->uuid,"fedcba987654321fedcba987654321fe");
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c:687]: Sonal agreed with this right? Please double check with @maxzhen, this is the first time we hardcode uuid inside driver. We are ok with hardcode configurations, but this seems tied with one specific shell, if you upgrade the shell, driver will be broken. Why do we push this code into XRT master?

Comment (maxzhen) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c:687]: not sure why we want to hard code a uuid?

Comment (sonals) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c:687]: Can this UUID be passed as a module param which will force users to know what they are doing?
Also please add a note in the code above that UUIDs are best obtained via VSEC and this side band is for testing.

**Review by sonals — CHANGES_REQUESTED** [review]
Comments inline

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_xclbin.c [hunk 3]
```diff
@@ -32,8 +32,13 @@ static int versal_xclbin_pre_download(xdev_handle_t xdev, void *args)
 	int ret = 0;
 
 	ret = xrt_xclbin_get_section(xclbin, PARTITION_METADATA, &metadata, &size);
-	if (ret)
-		return ret;
+	if (ret) {
+		/* PARTITION_METADATA is not present for FLAT shells */
+		if (xclbin->m_header.m_mode == XCLBIN_FLAT) {
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_xclbin.c:37]: the check should be moved to line 33

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c (hunk 2 above)
Comment (chvamshi-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c:687]: Hi @houlz0507 / @maxzhen  / @xdavidz, Sent a reply via email about the background of this change.

Sure @sonals , We can add a module param for this. We will also add a comment.

**Review by maxzhen — APPROVED** [review]

**Review by xdavidz — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c (hunk 1 above)
Comment (rbramand-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c:1125]: VERSAL flag is needed for downloading xclbin and for flashing shell, so added both flags.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c (hunk 2 above)
Comment (rbramand-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c]: @sonals added changes to pass uuid as module param, please review

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_xclbin.c (hunk 3 above)
Comment (rbramand-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/xocl_xclbin.c]: @houlz0507 made changes as requested

**Review by houlz0507 — APPROVED** [review]

**Pull request merged**